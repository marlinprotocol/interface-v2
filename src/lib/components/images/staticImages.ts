const Alert = '/images/alert.svg';
const Arrow = '/images/arrowicon.svg';
const ArrowLong = '/images/arrow-long.svg';
const ArrowRightSolid = '/images/arrowRightSolid.svg';
const ArrowLeftSolid = '/images/arrowLeftSolid.svg';
const BlueAlert = '/images/blue-alert.svg';
const Check = '/images/vectorcheck.svg';
const CloseCircle = '/images/close-circle-icon.svg';
const Copy = '/images/copyicon.svg';
const CopyGrey = '/images/copy-btn.svg';
const Cycle = '/images/cycleimg.svg';
const CycleLoader = '/images/cycleloader.svg';
const History = '/images/historyicon.svg';
const Info = '/images/info.svg';
const LeftBlueArrow = '/images/left-blue-arrow.svg';
const Lock = '/images/lockicon.svg';
const Metamask = '/images/metamaskicon.svg';
const OpenInNew = '/images/openinnew.svg';
const RightArrow = '/images/rightarrow.svg';
const RightArrowWhite = '/images/rightarrowwhite.svg';
const Sheild = '/images/shield.svg';
const Sort = '/images/sort.svg';
const Timer = '/images/timerclock.svg';
const UnActiveTick = '/images/unactivetick.svg';
const Verified = '/images/verifiedicon.svg';
const WalletConnect = '/images/walleticon.svg';
const WalletConnected = '/images/wallet-connected.svg';
const PolygonLogo = '/images/polygon.svg';
const PolygonFullLogo = '/images/polygon-full.webp';
const ArbitrumLogo = '/images/arbitrum.webp';
const EthereumLogo = '/images/eth.webp';
const LineaLogo = '/images/linea.webp';
const MarlinLogo = 'https://blog.marlin.org/wp-content/uploads/2024/01/200x200.png';
const navButton = '/images/navbtn.svg';
const chainLogo = '/images/chainSwitchLogo.svg';
const discChainLog = '/images/discChain.svg';
const bridgeIcon = '/images/bridgeIcon.svg';
const bridgeIconBlue = '/images/bridgeIconBlue.svg';
const dashboardIcon = '/images/dashboardIcon.svg';
const dashboardIconBlue = '/images/dashboardIconBlue.svg';
const ecosystemIcon = '/images/ecosystemIcon.svg';
const ecosystemIconBlue = '/images/ecosystemIconBlue.svg';
const faqsIcon = '/images/faqsIcon.svg';
const faqsIconBlue = '/images/faqsIconBlue.svg';
const governanceIcon = '/images/governanceIcon.svg';
const governanceIconBlue = '/images/governanceIconBlue.svg';
const kalypsoIcon = '/images/kalypsoIcon.svg';
const kalypsoIconBlue = '/images/kalypsoIconBlue.svg';
const oysterIcon = '/images/oysterIcon.svg';
const oysterIconBlue = '/images/oysterIconBlue.svg';
const relayIcon = '/images/relayIcon.svg';
const relayIconBlue = '/images/relayIconBlue.svg';
const dataTransferIcon = '/images/dataTransferIcon.svg';
const dataTransferIconBlue = '/images/dataTransferIconBlue.svg';
const marlinSmLogo = '/images/marlinSmLogo.svg';
const marlinLgLogo = '/images/marlinLgLogo.svg';
const alertV2Icon = '/images/alertV2Icon.svg';
const sortV2Icon = '/images/sortV2Icon.svg';
const lightModeLogo = '/images/lightModeLogo.svg';
const developerChatIcon = '/images/developerChatIcon.svg';
const earthIcon = '/images/earth.svg';
const blogIcon = '/images/blog.svg';
const chevronDownIcon = '/images/chevronDownIcon.svg';
const arrowConversion = '/images/arrowConversion.svg';
const closeIcon = '/images/closeIcon.svg';
const successIcon = '/images/SuccessIcon.svg';
const alertRed = '/images/alertRed.svg';
const backIcon = '/images/backV2Icon.svg';

// history pages
const exchangeIcon = '/images/exchangeV2Icon.svg';
const cancelIcon = '/images/cancelV2Icon.svg';
const historyIcon = '/images/historyV2Icon.svg';
const conversionCycleIcon = '/images/conversionCycleV2Icon.svg';
const toastSuccessAlert = '/images/toastSuccessAlert.svg';
const toastErrorAlert = '/images/toastErrorAlert.svg';
const toastWarningAlert = '/images/toastWarningAlert.svg';
const conversionIcon = '/images/conversionIcon.svg';
const walleticonOyster = '/images/walletIconOyster.svg';
const clockIcon = '/images/clockIcon.svg';
const checkmark = '/images/checkmark.svg';
const tick = '/images/tick.svg';

const externalLinkIcon = '/images/externalLinkV2Icon.svg';
const searchV2Icon = '/images/searchV2Icon.svg';
const refreshV2Icon = '/images/refreshV2Icon.svg';
const infoV2Icon = '/images/infoV2Icon.svg';
const ankrLogo = '/images/ankr-blue-logo.webp';
const cradlesLogo = '/images/CradlesLogo.webp';
const brightlyStakeLogo = '/images/Brightlystakelogo.webp';
const midaSwapLogo = '/images/MidaswapLogo.webp';
const aspectaLogo = '/images/AspectaLogo.webp';
const echolinkLogo = '/images/EcholinkLogo.webp';
const publicAiLogo = '/images/PublicAIDarkLogo.webp';
const operatorLogo = '/images/operator.svg';
const sideLogo = '/images/side.svg';
const injectiveLogo = '/images/injective.webp';
const mellowLogo = '/images/mellow.webp';
const FlashBotsLogo = '/images/FlashBotsLogo.webp';
const gearBoxLogo = '/images/gearbox.webp';
const decimalLogo = '/images/decimal.webp';
const zkbobLogo = '/images/zkbob.svg';
const p2pLogo = '/images/p2p_logo_white.webp';
const t2t2Logo = '/images/T2T2Logo.png';
const hashkeyCloudLogo = '/images/HashkeyCloudLogo.webp';
const nodeasyLogo = '/images/NodeEasyLogo.webp';
const chorusOneLogo = '/images/ChorusOneLogo.webp';
const availLogo = '/images/availLogo.webp';
const stafiLogo = '/images/StafiLogo.webp';
const sjtUniversityLogo = '/images/sjtUni.webp';
const goSleepLogo = '/images/GoSleepLogo.webp';
const infstones = '/images/InfstonesLogo.webp';
const staking4allLogo = '/images/Staking4AllLogo.webp';
const degoLogo = '/images/DegoFinance.webp';
const promLogo = '/images/PromLogo.png';
const calendarIcon = '/images/calendar.svg';
const checkmarkCircleIcon = '/images/checkmark-circle.svg';
const clock = '/images/clock.svg';
const loadingIcon = '/images/loading.svg';
const yellowInfo = '/images/information-circle.svg';
const redAlert = '/images/alert-red.svg';
const greenTick = '/images/checkmark-circle-green.svg';
const deployIcon = '/images/deployIcon.svg';
const arrowUpRight = '/images/arrow-up-right.svg';
const backendServer = '/images/backend-servers.svg';
const chains = '/images/chains.svg';
const frontends = '/images/frontends.svg';
const instances = '/images/instances.svg';
const llms = '/images/llms.svg';
const regions = '/images/regions.svg';
const dnsLogo = '/images/3dns.svg';
const bonfidaLogo = '/images/bonfida.svg';
const spaceIdLogo = '/images/spaceid.svg';
const exploreIcon = '/images/explore.svg';
const copyIcon = '/images/copyIconNew.svg';
const infoIconPurple = '/images/information-purple.svg';

export const staticImages = {
	Alert,
	Arrow,
	ArrowLong,
	ArrowRightSolid,
	ArrowLeftSolid,
	BlueAlert,
	Check,
	CloseCircle,
	Copy,
	CopyGrey,
	Cycle,
	CycleLoader,
	History,
	Info,
	LeftBlueArrow,
	Lock,
	Metamask,
	OpenInNew,
	RightArrow,
	RightArrowWhite,
	Sheild,
	Sort,
	Timer,
	UnActiveTick,
	Verified,
	WalletConnect,
	WalletConnected,
	PolygonLogo,
	PolygonFullLogo,
	ArbitrumLogo,
	EthereumLogo,
	LineaLogo,
	MarlinLogo,
	navButton,
	chainLogo,
	discChainLog,
	bridgeIcon,
	bridgeIconBlue,
	dashboardIcon,
	dashboardIconBlue,
	dataTransferIcon,
	dataTransferIconBlue,
	ecosystemIcon,
	ecosystemIconBlue,
	faqsIcon,
	faqsIconBlue,
	governanceIcon,
	governanceIconBlue,
	kalypsoIcon,
	kalypsoIconBlue,
	oysterIcon,
	oysterIconBlue,
	relayIcon,
	relayIconBlue,
	marlinSmLogo,
	marlinLgLogo,
	alertV2Icon,
	sortV2Icon,
	lightModeLogo,
	developerChatIcon,
	earthIcon,
	blogIcon,
	chevronDownIcon,
	closeIcon,
	successIcon,
	alertRed,
	backIcon,
	exchangeIcon,
	cancelIcon,
	historyIcon,
	conversionCycleIcon,
	toastSuccessAlert,
	toastErrorAlert,
	toastWarningAlert,
	arrowConversion,
	conversionIcon,
	walleticonOyster,
	clockIcon,
	externalLinkIcon,
	checkmark,
	tick,
	searchV2Icon,
	refreshV2Icon,
	infoV2Icon,
	ankrLogo,
	cradlesLogo,
	brightlyStakeLogo,
	midaSwapLogo,
	aspectaLogo,
	echolinkLogo,
	publicAiLogo,
	operatorLogo,
	sideLogo,
	injectiveLogo,
	mellowLogo,
	FlashBotsLogo,
	gearBoxLogo,
	decimalLogo,
	zkbobLogo,
	p2pLogo,
	t2t2Logo,
	hashkeyCloudLogo,
	nodeasyLogo,
	chorusOneLogo,
	availLogo,
	stafiLogo,
	sjtUniversityLogo,
	goSleepLogo,
	infstones,
	staking4allLogo,
	degoLogo,
	promLogo,
	calendarIcon,
	checkmarkCircleIcon,
	clock,
	loadingIcon,
	yellowInfo,
	redAlert,
	greenTick,
	deployIcon,
	backendServer,
	chains,
	frontends,
	instances,
	llms,
	regions,
	arrowUpRight,
	dnsLogo,
	bonfidaLogo,
	spaceIdLogo,
	exploreIcon,
	copyIcon,
	infoIconPurple
};
