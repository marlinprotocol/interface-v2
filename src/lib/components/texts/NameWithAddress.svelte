<script lang="ts">
	import Text from '$lib/atoms/texts/Text.svelte';
	import Tooltip from '$lib/atoms/tooltips/Tooltip.svelte';
	import { addToast } from '$lib/data-stores/toastStore';
	import { shortenText } from '$lib/utils/helpers/conversionHelper';
	import { copyTextToClipboard } from '$lib/utils/helpers/commonHelper';

	export let name = '';
	export let address = '';
	export let long: boolean = false;
	export let table:
		| 'inventory'
		| 'inventory-history'
		| 'marketplace'
		| 'operator'
		| 'operator-hitory' = 'marketplace';

	const isOneOfInventoryTables = table === 'inventory' || table === 'inventory-history';

	const onCopyAddress = () => {
		copyTextToClipboard(address);
		addToast({
			message: {
				title: 'Copied',
				description: `${isOneOfInventoryTables ? 'Id' : 'Address'} copied to clipboard`
			},
			variant: 'success'
		});
	};

	let addressText = shortenText(address, long ? 10 : 6, long ? 10 : 6);
</script>

<div
	class="ml-4 flex w-fit items-center justify-start"
	data-testid="name-with-address"
	id="name-with-address"
>
	<div class="flex flex-col items-start">
		{#if name}
			<Tooltip placement="right">
				<span slot="tooltipContent">{name}</span>
				<Text
					slot="tooltipIcon"
					variant="mini"
					fontWeight="font-normal"
					text={name.length > 13 ? name.slice(0, 13).trim() + '...' : name}
					styleClass="truncate w-min"
				/>
			</Tooltip>
			<div class="mt-1 flex w-fit items-center gap-1">
				<Text
					variant="tiny"
					styleClass="text-grey"
					fontWeight="font-normal"
					bind:text={addressText}
				/>
				<button on:keypress={onCopyAddress} on:click={onCopyAddress}>
					<slot name="copyIcon" />
				</button>
			</div>
		{:else}
			<div class="flex w-fit min-w-[150px] items-center gap-1">
				<Text variant="nav" fontWeight="font-normal" text={addressText} />
				<button on:keypress={onCopyAddress} on:click={onCopyAddress}>
					<slot name="copyIcon" />
				</button>
			</div>
		{/if}
	</div>
</div>
