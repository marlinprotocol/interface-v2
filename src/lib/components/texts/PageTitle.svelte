<script lang="ts">
	import Text from '$lib/atoms/texts/Text.svelte';
	import BackButton from '$lib/components/buttons/BackButton.svelte';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let title: string;
	export let backHref = '';
	export let wrapperClass = '';
</script>

<div class={cn('mb-8 flex items-center gap-4', wrapperClass)} data-testid="page-title">
	{#if !!backHref}
		<BackButton href={backHref} text="" />
	{/if}
	<Text
		styleClass="font-poppins leading-[-2px] text-grey-800"
		variant="h2"
		fontWeight="font-medium"
		text={title}
	/>
	<slot name="button" />
</div>
