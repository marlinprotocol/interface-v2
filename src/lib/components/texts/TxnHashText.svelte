<script lang="ts">
	import { shortenText } from '$lib/utils/helpers/conversionHelper';
	import TxnIcon from '$lib/components/icons/TxnIcon.svelte';
	export let txnHash: string;
	export let txnHashUrl: string;
	export let startInt = 6;
	export let endInt = 3;
</script>

<div class="flex items-center justify-start gap-2" data-testid="txn-hash-text">
	{shortenText(txnHash, startInt, endInt)}
	<TxnIcon {txnHashUrl} />
</div>
