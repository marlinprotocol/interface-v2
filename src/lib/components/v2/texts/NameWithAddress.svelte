<script lang="ts">
	import Text from '$lib/atoms/v2/texts/Text.svelte';
	import Tooltip from '$lib/atoms/v2/tooltips/Tooltip.svelte';
	import { addToast } from '$lib/data-stores/v2/toastStore';
	import { shortenText } from '$lib/utils/helpers/conversionHelper';
	import { copyTextToClipboard } from '$lib/utils/helpers/commonHelper';
	import { getColorHexForTableRow } from '$lib/utils/v2/helpers/componentHelper';

	export let name = '';
	export let address = '';
	export let rowIndex = -1;

	const onCopyAddress = () => {
		copyTextToClipboard(address);
		addToast({
			message: {
				title: 'Copied',
				description: `Address copied to clipboard`
			},
			variant: 'success'
		});
	};

	//get first 2 letters of name or address if name is empty
	const startLetters = name ? name.slice(0, 2) : address.slice(0, 2);
	const bgColor = getColorHexForTableRow(rowIndex);
</script>

<div
	class="ml-4 flex w-fit items-center justify-start"
	data-testid="name-with-address"
	id="name-with-address"
>
	<div class="flex flex-col items-start">
		{#if name}
			<Tooltip>
				<span slot="tooltipContent">{name}</span>
				<Text
					slot="tooltipIcon"
					variant="mini"
					fontWeight="font-normal"
					text={name.length > 13 ? name.slice(0, 13).trim() + '...' : name}
					styleClass="truncate w-min"
				/>
			</Tooltip>
			<div class="mt-1 flex w-fit items-center gap-1">
				<Text
					variant="tiny"
					styleClass="text-grey"
					fontWeight="font-normal"
					text={shortenText(address, 6, 6)}
				/>
				<button on:keypress={onCopyAddress} on:click={onCopyAddress}>
					<slot name="copyIcon" />
				</button>
			</div>
		{:else}
			<div class="flex w-fit min-w-[150px] items-center gap-1">
				<Text variant="nav" fontWeight="font-normal" text={shortenText(address, 6, 6)} />
				<button on:keypress={onCopyAddress} on:click={onCopyAddress}>
					<slot name="copyIcon" />
				</button>
			</div>
		{/if}
	</div>
</div>
