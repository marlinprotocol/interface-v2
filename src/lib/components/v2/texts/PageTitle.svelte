<script lang="ts">
	import Text from '$lib/atoms/v2/texts/Text.svelte';
	import BackButton from '$lib/components/v2/buttons/BackButton.svelte';

	export let title: string;
	export let backHref = '';
</script>

<div class="mb-8 flex items-center gap-4" data-testid="page-title">
	{#if !!backHref}
		<BackButton href={backHref} text="" />
	{/if}
	<Text
		styleClass="font-poppins leading-[-2px] text-[#030115]"
		variant="h2"
		fontWeight="font-medium"
		text={title}
	/>
	<slot name="button" />
</div>
