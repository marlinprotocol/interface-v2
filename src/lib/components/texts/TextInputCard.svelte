<script lang="ts">
	import InputCard from '$lib/atoms/cards/InputCard.svelte';
	import Text from '$lib/atoms/texts/Text.svelte';
	import TooltipIcon from '$lib/atoms/tooltips/TooltipIcon.svelte';
	import { handleCopyClick } from '$lib/utils/helpers/componentHelper';
	import { staticImages } from '$lib/components/images/staticImages';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let title: string;
	export let tooltipText = '';
	export let value = '';
	export let textStyle = '';
	export let centered = false;
	export let cliboardContent: string | undefined = '';

	$: successMessage = `${title} copied to clipboard`;
</script>

<InputCard variant="v2Input">
	<div
		class={cn('mb-1 flex items-center gap-1', {
			'justify-center': centered
		})}
	>
		<Text
			variant="small"
			text={title}
			styleClass="absolute left-[18px] top-[-10px] mb-4 bg-white  font-poppins text-sm font-light text-[#030115]"
		/>
		{#if tooltipText}
			<TooltipIcon
				{tooltipText}
				tooltipVariant="tooltip-secondary"
				tooltipDirection="tooltip-right"
			/>
		{/if}
	</div>
	<div
		class={cn('flex', textStyle, {
			'items-center justify-center text-center': centered
		})}
	>
		<Text
			variant="body"
			text={value}
			fontWeight="font-normal"
			styleClass={cn('flex py-[3px]', textStyle, {
				'items-center justify-center text-center': centered
			})}
		/>
		{#if cliboardContent !== '' && cliboardContent !== undefined}
			<button
				id="text-input-card-button"
				class="ml-2 w-6 cursor-pointer"
				on:click={() => {
					handleCopyClick(cliboardContent, successMessage);
				}}
			>
				<img src={staticImages.copyIcon} alt="Copy" />
			</button>
		{/if}
	</div>
</InputCard>
