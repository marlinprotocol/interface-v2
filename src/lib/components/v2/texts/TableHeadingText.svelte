<script lang="ts">
	import type { TableModel } from '$lib/types/v2/componentTypes';
	import { staticImages } from '$lib/components/images/staticImages';
	import Tooltip from '$lib/atoms/v2/tooltips/Tooltip.svelte';
	import type { TooltipPlacement } from '$lib/types/v2/componentTypes';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let heading: TableModel['header'];
	export let placement: TooltipPlacement = 'right';
	export let styleClass = '';
	export let iconWidth = '16px';
	export let handleSortData: ((id: string) => void) | undefined = undefined;

	const { title = '', tooltipText = '', sorting = false } = heading;
</script>

<div
	class={cn(
		'mt-0.5 flex flex-nowrap items-center justify-start gap-1 text-center text-[14px] font-normal text-[#030115]',
		styleClass
	)}
	data-testid="table-heading-text"
>
	{#if sorting}
		<button
			class="relative flex items-center gap-0.5"
			on:click={() => handleSortData?.(heading.id)}
		>
			<img class="absolute left-[-5px]" src={staticImages.sortV2Icon} alt="sort" width="16px" />
			<span class="w-fit pl-4 tracking-widest">{title}</span>
			{#if tooltipText !== ''}
				<Tooltip {placement}>
					<img slot="tooltipIcon" src={staticImages.alertV2Icon} alt="Info" width={iconWidth} />
					<span slot="tooltipContent">{tooltipText}</span></Tooltip
				>
			{/if}
		</button>
	{:else}
		<div class="relative flex items-center gap-1">
			<span class="w-fit tracking-widest">{title}</span>
			{#if tooltipText !== ''}
				<Tooltip {placement}>
					<img slot="tooltipIcon" src={staticImages.alertV2Icon} alt="Info" width={iconWidth} />
					<span slot="tooltipContent">{tooltipText}</span></Tooltip
				>
			{/if}
		</div>
	{/if}
</div>
