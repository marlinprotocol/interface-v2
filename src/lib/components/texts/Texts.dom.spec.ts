import { cleanup, fireEvent, render } from '@testing-library/svelte';
import NameWithAddress from './NameWithAddress.svelte';
import PageTitle from './PageTitle.svelte';
import TableHeadingText from './TableHeadingText.svelte';
import TextInputCard from './TextInputCard.svelte';
import TxnHashText from './TxnHashText.svelte';
import { staticImages } from '$lib/components/images/staticImages';

describe('Text', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
	});

	it('renders NameWithAddress with default props', () => {
		const { getByTestId, container } = render(NameWithAddress);
		expect(
			getByTestId('name-with-address').firstElementChild?.getElementsByTagName('span')[0]
		).toBeFalsy();
		expect(getByTestId('name-with-address').id === 'name-with-address').toBe(true);
		expect(container).toMatchSnapshot();
	});

	it('renders NameWithAddress with custom props', () => {
		const { getByTestId } = render(NameWithAddress, {
			props: { address: 'xyz street , city:abc', name: 'Robert', rowIndex: 2 }
		});
		expect(getByTestId('name-with-address').id === 'name-with-address').toBe(true);
		expect(
			getByTestId('name-with-address').firstElementChild?.firstElementChild?.getElementsByTagName(
				'span'
			)[0].innerHTML === 'Robert'
		).toBe(true);

		expect(
			getByTestId('name-with-address').lastElementChild?.firstElementChild?.getElementsByTagName(
				'span'
			)[0].innerHTML === 'Robert'
		).toBe(true);
	});

	it('renders PageTitle with default props', () => {
		const { getByTestId, container } = render(PageTitle, { props: { title: '' } });
		expect(getByTestId('page-title').contains(getByTestId('text'))).toBe(true);
		expect(getByTestId('text').innerHTML).toBeFalsy();
		expect(container).toMatchSnapshot();
	});

	it('renders PageTitle with props', () => {
		const { getByTestId } = render(PageTitle, {
			props: { title: 'Test Page', backHref: 'https://test.com' }
		});
		expect(getByTestId('page-title').contains(getByTestId('text'))).toBe(true);
		expect(getByTestId('page-title').contains(getByTestId('back-button'))).toBe(true);
		expect(getByTestId('text').innerHTML === 'Test Page').toBeTruthy();
	});

	it('renders TableHeadingText with default props', () => {
		const { getByTestId, container } = render(TableHeadingText, {
			props: { heading: { id: '', title: '' } }
		});
		expect(
			getByTestId('table-heading-text').firstElementChild?.firstElementChild?.classList.contains(
				'tracking-widest'
			)
		).toBe(true);
		expect(
			getByTestId('table-heading-text').firstElementChild?.firstElementChild?.innerHTML
		).toBeFalsy();
		expect(container).toMatchSnapshot();
	});

	it('renders TableHeadingText with custom props', () => {
		const { getByTestId, getByAltText } = render(TableHeadingText, {
			props: {
				heading: {
					id: 'test-table-heading-text',
					title: 'Table Heading',
					tooltipText: 'Table Heading Tooltip',
					sorting: true
				},
				iconWidth: '14px',
				styleClass: 'additional-child-class'
			}
		});
		expect(getByTestId('table-heading-text').contains(getByAltText('sort'))).toBeTruthy();
		expect(getByTestId('table-heading-text').contains(getByAltText('Info'))).toBeTruthy();
		expect(getByAltText('sort').getAttribute('src') === staticImages.sortV2Icon).toBeTruthy();
		expect(getByAltText('Info').getAttribute('src') === staticImages.alertV2Icon).toBeTruthy();
	});

	it('renders TextInputCard with default props', () => {
		const { getByTestId, getAllByTestId, container } = render(TextInputCard, {
			props: { title: '' }
		});
		expect(getByTestId('input-card').contains(getAllByTestId('text')[0])).toBeTruthy();
		expect(getByTestId('input-card').contains(getAllByTestId('text')[1])).toBeTruthy();
		expect(getAllByTestId('text')[0].innerHTML).toBeFalsy();
		expect(getAllByTestId('text')[1].innerHTML).toBeFalsy();
		expect(container).toMatchSnapshot();
	});

	it('renders TextInputCard with custom props', () => {
		const { getByTestId, getAllByTestId } = render(TextInputCard, {
			props: {
				title: 'Text input card title',
				value: 'Text input card value',
				centered: true
			}
		});
		expect(getByTestId('input-card').contains(getAllByTestId('text')[0])).toBeTruthy();
		expect(getByTestId('input-card').contains(getAllByTestId('text')[1])).toBeTruthy();
		expect(getAllByTestId('text')[0].innerHTML === 'Text input card title').toBeTruthy();
		expect(getAllByTestId('text')[1].innerHTML === 'Text input card value').toBeTruthy();
		expect(getByTestId('input-card').firstElementChild?.classList.contains('justify-center')).toBe(
			true
		);
		expect(getByTestId('input-card').lastElementChild?.classList.contains('justify-center')).toBe(
			true
		);
	});

	it('renders TextInputCard with cliboardContent props', async () => {
		const { getByTestId, getAllByTestId, getByAltText } = render(TextInputCard, {
			props: {
				title: 'Text input card title',
				cliboardContent: 'copy clipbard content'
			}
		});
		expect(getByTestId('input-card').contains(getAllByTestId('text')[0])).toBeTruthy();
		expect(getAllByTestId('text')[0].innerHTML === 'Text input card title').toBeTruthy();
		const button = getByAltText('Copy').parentElement as Element;
		expect(button.id).toBeTruthy();
		await fireEvent.click(button);
		expect(button.id === 'text-input-card-button').toBeTruthy();
	});

	it('renders TxnHashText with default props', () => {
		const { getByTestId, getByAltText, container } = render(TxnHashText, {
			props: { txnHash: '', txnHashUrl: '' }
		});
		expect(getByTestId('txn-hash-text').contains(getByAltText('txn link'))).toBe(true);
		expect(getByAltText('txn link').getAttribute('src') === staticImages.OpenInNew).toBe(true);
		expect(getByAltText('txn link').parentElement?.getAttribute('href')).toBeFalsy();
		expect(container).toMatchSnapshot();
	});

	it('renders TxnHashText with custom props', () => {
		const { getByTestId, getByAltText } = render(TxnHashText, {
			props: {
				txnHash: 'Text txnHash',
				txnHashUrl: 'https://test-txnhash.com',
				startInt: 4,
				endInt: 3
			}
		});
		expect(getByTestId('txn-hash-text').contains(getByAltText('txn link'))).toBe(true);
		expect(getByAltText('txn link').getAttribute('src') === staticImages.OpenInNew).toBe(true);
		expect(
			getByAltText('txn link').parentElement?.getAttribute('href') === 'https://test-txnhash.com'
		).toBe(true);
		expect(getByTestId('txn-hash-text').innerHTML.includes('Text...ash')).toBe(true);
	});
});
