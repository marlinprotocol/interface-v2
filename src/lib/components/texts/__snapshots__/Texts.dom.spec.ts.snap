// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > renders NameWithAddress with default props 1`] = `
<body>
  <div>
    <div
      class="ml-4 flex w-fit items-center justify-start"
      data-testid="name-with-address"
      id="name-with-address"
    >
      <div
        class="flex flex-col items-start"
      >
        <div
          class="flex w-fit min-w-[150px] items-center gap-1"
        >
          <div
            class=" text-sm font-normal"
            data-testid="text"
          >
            
          </div>
          <!--&lt;Text&gt;-->
           
          <button />
        </div>
      </div>
    </div>
    <!--&lt;NameWithAddress&gt;-->
  </div>
</body>
`;

exports[`Text > renders PageTitle with default props 1`] = `
<body>
  <div>
    <div
      class="mb-8 flex items-center gap-4"
      data-testid="page-title"
    >
       
      <div
        class="font-poppins leading-[-2px] text-[#030115] text-3xl font-medium"
        data-testid="text"
      >
        
      </div>
      <!--&lt;Text&gt;-->
       
    </div>
    <!--&lt;PageTitle&gt;-->
  </div>
</body>
`;

exports[`Text > renders TableHeadingText with default props 1`] = `
<body>
  <div>
    <div
      class="mt-0.5 flex flex-nowrap items-center justify-start gap-1 text-center text-[14px] font-normal text-[#030115]"
      data-testid="table-heading-text"
    >
      <div
        class="relative flex items-center gap-1"
      >
        <span
          class="w-fit tracking-widest"
        >
          
        </span>
         
      </div>
    </div>
    <!--&lt;TableHeadingText&gt;-->
  </div>
</body>
`;

exports[`Text > renders TextInputCard with default props 1`] = `
<body>
  <div>
    <div
      class="w-full relative border border-[#D9DADE] bg-white py-2.5 px-4 rounded-full"
      data-testid="input-card"
      id=""
    >
      <div
        class="mb-1 flex items-center gap-1"
      >
        <div
          class="absolute left-[18px] top-[-10px] mb-4 bg-white px-2 font-poppins text-sm font-light text-[#030115] text-sm font-normal"
          data-testid="text"
        >
          
        </div>
        <!--&lt;Text&gt;-->
         
      </div>
       
      <div
        class="flex"
      >
        <div
          class="flex py-[3px] text-base font-normal"
          data-testid="text"
        >
          
        </div>
        <!--&lt;Text&gt;-->
         
      </div>
    </div>
    <!--&lt;InputCard&gt;-->
    <!--&lt;TextInputCard&gt;-->
  </div>
</body>
`;

exports[`Text > renders TxnHashText with default props 1`] = `
<body>
  <div>
    <div
      class="flex items-center justify-start gap-2"
      data-testid="txn-hash-text"
    >
      
       
      <a
        class="shrink-0"
        href=""
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="txn link"
          src="/images/openinnew.svg"
          width="13px"
        />
      </a>
      <!--&lt;TxnIcon&gt;-->
    </div>
    <!--&lt;TxnHashText&gt;-->
  </div>
</body>
`;
